python/pyarrow/_flight.pyx

```diff
@@ -1561,9 +1583,14 @@ cdef CStatus _do_action_result_next(
 
     try:
         action_result = next(<object> self)
-        if not isinstance(action_result, Result):
+        if isinstance(action_result, str):
+            action_result = Result(action_result.encode('utf-8'))
```

Comment:
Not sure this is desirable. An action result is an arbitrary bytestring.

Reply:
I see. I'll remove this, then. I think the requirement should be that results are either `Result` or something exporting the buffer protocol